### PR TITLE
Add RFP (draft) from OSTIF for third-party security audit of Kubernetes in 2024

### PR DESCRIPTION
**Pull request by reylejano**
This PR adds the RFP (currently in draft) from the Open Source Improvement Fund (OSTIF) for the RFP for the third-party security audit of Kubernetes in 2024.
This is a mirror of the Google Doc from OSTIF: https://docs.google.com/document/d/10Qepg6Tb9Qwkb4dy0lhHD0k0ZKw6z-PRD7z2dNPEJBo/edit?tab=t.0

/hold
Although in draft, we welcome comments, suggestions, and reviews

### CODE REVIEW

**Comment by tabbysable** [comment]
Link to view it rendered, for convenience: https://github.com/reylejano/sig-security/blob/draft-ostif-rfp-2024/sig-security-external-audit/security-audit-2024/RFP.md

File: sig-security-external-audit/security-audit-2024/RFP.md [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+
+Below are the components identified by the Kubernetes community as both needing additional security scrutiny and whose maintainers are open to enthusiastically participating in this project.
+
+These projects are ranked by importance on both the perceived criticality of the component by the community, and the information level of the documentation provided. It should also be carefully noted that auditors will have the flexibility to investigate hunches and seek out issues in other components if they so choose. The prioritization listed here is only compulsory for components specifically named below, or projects listed as “medium” and “high” priority in the [2024 Kubernetes Audit Scope document](https://docs.google.com/spreadsheets/d/1b_OrTHzLcPisShcera9eKtycKZORvW1z8Hm2If4NCF4).
```
Comment (tabbysable) [inline sig-security-external-audit/security-audit-2024/RFP.md:136]: What exactly is meant by "The prioritization listed here is only compulsory for components specifically named below?" There is a list of 19 components that follows.

Is it expected that each proposal's scope be a subset of this list starting at the top and working its way down?

Are vendors free to choose a subset of these 19 components to scope their proposal, based on the expertise and time commitment available among their staff?

Comment (OSTIF-Derek) [reply sig-security-external-audit/security-audit-2024/RFP.md:136]: I can reword this section to get the intent across. Basically we think that there is not enough time available to the researchers to get to all 90+ items, especially considering some of them are large and very complex. This wording is to prioritize the specific items named, but for all other subprojects/features it will be a best-effort approach to get as far as you can, and also investigate the components that the researchers find the most interesting. It also calls for the researchers to carefully call out what was covered by this review, and to specifically call out what they couldn't get to. This helps future independent researchers know where eyes have been and what hasn't been looked at in any follow-up work.

I'll reword this to be more clear.

Comment (tabbysable) [reply sig-security-external-audit/security-audit-2024/RFP.md:136]: thank you! That'll be really helpful, I think.

The previous RFP may be useful as a source of inspiration and wording: https://github.com/kubernetes/sig-security/blob/main/sig-security-external-audit/security-audit-2021-2022/RFP.md

Comment (OSTIF-Derek) [reply sig-security-external-audit/security-audit-2024/RFP.md:136]: Ack

I will review it while rewording that section.

Thank you!

Comment (reylejano) [reply sig-security-external-audit/security-audit-2024/RFP.md]: @tabbysable , I've updated this PR with updates from @OSTIF-Derek

**Comment by tabbysable** [comment]
This is the text of the RFP as was published by OSTIF, and responses have been received. Getting this merged for future reference.

/unhold
/lgtm
